### DISCUSSION TITLE
ESP32S3 with W5500 build

### DISCUSSION THREAD

**Discussion by MikePolynin** [micropython/micropython#12820] [ESP32] [answered]
Hello everyone

I`m trying to build a micropython firmware for ESP32S3 32Mb with W5500 ethernet module but it`s going wrong.
Build with ESP-IDF 5.1.1 and with the latest micropython from Git repo.

Turn on W5500 module support thru 'component config -> ethernet -> support spi to ethernet module' in idf.py menuconfig.
And when i run make command after i get message:
_HINT: Macro ETH_W5500_DEFAULT_CONFIG now accepts both SPI host and SPI device related configuration. The SPI-Ethernet Module initialization has been simplified to allocate an SPI device internally so the configuration structure requires the related configuration.
Please refer to the Networking migration guide, section SPI-Ethernet Module Initialization, for more details._
Without W5500 module support building is ok.

What shall i do? What`s wrong with settings?

**Comment by robert-hh** [comment]
Look at PR https://github.com/micropython/micropython/pull/12736, which re-enables support for the W5500 with esp-idf v5.x. Tested with a generic ESP32 device and esp-idf v5.0.2. The changes are not device-specific. So it should work with an esp32S3 as well. esp-idf v5.0.2 is the version used for MicroPython at the moment. Building with v5.1.1 may work, but is not tested yet.

**Reply by MikePolynin** [reply]
Heloo @robert-hh 

Thanks a lot, it was helpful but not at all
There is the listing with warnings and errors after 'idf.py build' command:

> In file included from /home/mickey-p/micropython/ports/esp32/managed_components/espressif__tinyusb/src/osal/osal_freertos.h:33,
>                  from /home/mickey-p/micropython/ports/esp32/managed_components/espressif__tinyusb/src/osal/osal.h:57,
>                  from /home/mickey-p/micropython/ports/esp32/managed_components/espressif__tinyusb/src/tusb.h:38,
>                  from /home/mickey-p/micropython/ports/esp32/managed_components/espressif__esp_tinyusb/include/tinyusb.h:11,
>                  from /home/mickey-p/micropython/ports/esp32/usb.c:35:
> /home/mickey-p/micropython/lib/berkeley-db-1.xx/PORT/include/queue.h:85: warning: "LIST_INIT" redefined
>    85 | #define LIST_INIT(head) {                                               \
>       | 
> In file included from /home/mickey-p/esp-idf/components/esp_wifi/include/esp_private/esp_wifi_types_private.h:20,
>                  from /home/mickey-p/esp-idf/components/esp_wifi/include/esp_wifi_types.h:11,
>                  from /home/mickey-p/micropython/ports/esp32/mpconfigport.h:13,
>                  from /home/mickey-p/micropython/py/mpconfig.h:91,
>                  from /home/mickey-p/micropython/py/mpstate.h:31,
>                  from /home/mickey-p/micropython/py/runtime.h:29,
>                  from /home/mickey-p/micropython/ports/esp32/usb.c:27:
> /home/mickey-p/.espressif/tools/xtensa-esp32s3-elf/esp-12.2.0_20230208/xtensa-esp32s3-elf/xtensa-esp32s3-elf/sys-include/sys/queue.h:563: note: this is the location of the previous definition
>   563 | #define LIST_INIT(head) do {                                            \
>       | 
> /home/mickey-p/micropython/lib/berkeley-db-1.xx/PORT/include/queue.h:89: warning: "LIST_INSERT_AFTER" redefined
>    89 | #define LIST_INSERT_AFTER(listelm, elm, field) {                        \
>       | 
> /home/mickey-p/.espressif/tools/xtensa-esp32s3-elf/esp-12.2.0_20230208/xtensa-esp32s3-elf/xtensa-esp32s3-elf/sys-include/sys/queue.h:567: note: this is the location of the previous definition
>   567 | #define LIST_INSERT_AFTER(listelm, elm, field) do {                     \
>       | 
> /home/mickey-p/micropython/lib/berkeley-db-1.xx/PORT/include/queue.h:97: warning: "LIST_INSERT_HEAD" redefined
>    97 | #define LIST_INSERT_HEAD(head, elm, field) {                            \
>       | 
> /home/mickey-p/.espressif/tools/xtensa-esp32s3-elf/esp-12.2.0_20230208/xtensa-esp32s3-elf/xtensa-esp32s3-elf/sys-include/sys/queue.h:584: note: this is the location of the previous definition
>   584 | #define LIST_INSERT_HEAD(head, elm, field) do {                         \
>       | 
> /home/mickey-p/micropython/lib/berkeley-db-1.xx/PORT/include/queue.h:104: warning: "LIST_REMOVE" redefined
>   104 | #define LIST_REMOVE(elm, field) {                                       \
>       | 
> /home/mickey-p/.espressif/tools/xtensa-esp32s3-elf/esp-12.2.0_20230208/xtensa-esp32s3-elf/xtensa-esp32s3-elf/sys-include/sys/queue.h:599: note: this is the location of the previous definition
>   599 | #define LIST_REMOVE(elm, field) do {                                    \
>       | 
> /home/mickey-p/micropython/lib/berkeley-db-1.xx/PORT/include/queue.h:114: warning: "TAILQ_HEAD" redefined
>   114 | #define TAILQ_HEAD(name, type)                                          \
>       | 
> /home/mickey-p/.espressif/tools/xtensa-esp32s3-elf/esp-12.2.0_20230208/xtensa-esp32s3-elf/xtensa-esp32s3-elf/sys-include/sys/queue.h:625: note: this is the location of the previous definition
>   625 | #define TAILQ_HEAD(name, type)                                          \
>       | 
> /home/mickey-p/micropython/lib/berkeley-db-1.xx/PORT/include/queue.h:120: warning: "TAILQ_ENTRY" redefined
>   120 | #define TAILQ_ENTRY(type)                                               \
>       | 
> /home/mickey-p/.espressif/tools/xtensa-esp32s3-elf/esp-12.2.0_20230208/xtensa-esp32s3-elf/xtensa-esp32s3-elf/sys-include/sys/queue.h:642: note: this is the location of the previous definition
>   642 | #define TAILQ_ENTRY(type)                                               \
>       | 
> /home/mickey-p/micropython/lib/berkeley-db-1.xx/PORT/include/queue.h:129: warning: "TAILQ_INIT" redefined
>   129 | #define TAILQ_INIT(head) {                                              \
>       | 
> /home/mickey-p/.espressif/tools/xtensa-esp32s3-elf/esp-12.2.0_20230208/xtensa-esp32s3-elf/xtensa-esp32s3-elf/sys-include/sys/queue.h:767: note: this is the location of the previous definition
>   767 | #define TAILQ_INIT(head) do {                                           \
>       | 
> /home/mickey-p/micropython/lib/berkeley-db-1.xx/PORT/include/queue.h:134: warning: "TAILQ_INSERT_HEAD" redefined
>   134 | #define TAILQ_INSERT_HEAD(head, elm, field) {                           \
>       | 
> /home/mickey-p/.espressif/tools/xtensa-esp32s3-elf/esp-12.2.0_20230208/xtensa-esp32s3-elf/xtensa-esp32s3-elf/sys-include/sys/queue.h:798: note: this is the location of the previous definition
>   798 | #define TAILQ_INSERT_HEAD(head, elm, field) do {                        \
>       | 
> /home/mickey-p/micropython/lib/berkeley-db-1.xx/PORT/include/queue.h:144: warning: "TAILQ_INSERT_TAIL" redefined
>   144 | #define TAILQ_INSERT_TAIL(head, elm, field) {                           \
>       | 
> /home/mickey-p/.espressif/tools/xtensa-esp32s3-elf/esp-12.2.0_20230208/xtensa-esp32s3-elf/xtensa-esp32s3-elf/sys-include/sys/queue.h:811: note: this is the location of the previous definition
>   811 | #define TAILQ_INSERT_TAIL(head, elm, field) do {                        \
>       | 
> /home/mickey-p/micropython/lib/berkeley-db-1.xx/PORT/include/queue.h:151: warning: "TAILQ_INSERT_AFTER" redefined
>   151 | #define TAILQ_INSERT_AFTER(head, listelm, elm, field) {                 \
>       | 
> /home/mickey-p/.espressif/tools/xtensa-esp32s3-elf/esp-12.2.0_20230208/xtensa-esp32s3-elf/xtensa-esp32s3-elf/sys-include/sys/queue.h:773: note: this is the location of the previous definition
>   773 | #define TAILQ_INSERT_AFTER(head, listelm, elm, field) do {              \
>       | 
> /home/mickey-p/micropython/lib/berkeley-db-1.xx/PORT/include/queue.h:161: warning: "TAILQ_REMOVE" redefined
>   161 | #define TAILQ_REMOVE(head, elm, field) {                                \
>       | 
> /home/mickey-p/.espressif/tools/xtensa-esp32s3-elf/esp-12.2.0_20230208/xtensa-esp32s3-elf/xtensa-esp32s3-elf/sys-include/sys/queue.h:839: note: this is the location of the previous definition
>   839 | #define TAILQ_REMOVE(head, elm, field) do {                             \
>       | 
> In file included from /home/mickey-p/micropython/ports/esp32/usb.c:36:
> /home/mickey-p/micropython/ports/esp32/managed_components/espressif__esp_tinyusb/include/tusb_cdc_acm.h:51:5: error: expected specifier-qualifier-list before 'cdc_line_coding_t'
>    51 |     cdc_line_coding_t const *p_line_coding; /*!< New line coding value */
>       |     ^~~~~~~~~~~~~~~~~
> /home/mickey-p/micropython/ports/esp32/usb.c:41:27: error: 'CONFIG_TINYUSB_CDC_RX_BUFSIZE' undeclared here (not in a function); did you mean 'CFG_TUD_CDC_RX_BUFSIZE'?
>    41 | static uint8_t usb_rx_buf[CONFIG_TINYUSB_CDC_RX_BUFSIZE];
>       |                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
>       |                           CFG_TUD_CDC_RX_BUFSIZE
> /home/mickey-p/micropython/ports/esp32/usb.c: In function 'usb_tx_strn':
> /home/mickey-p/micropython/ports/esp32/usb.c:92:12: error: implicit declaration of function 'tud_cdc_n_connected'; did you mean 'tud_connected'? [-Werror=implicit-function-declaration]
>    92 |     while (tud_cdc_n_connected(CDC_ITF) && len && esp_timer_get_time() < timeout) {
>       |            ^~~~~~~~~~~~~~~~~~~
>       |            tud_connected
> /home/mickey-p/micropython/ports/esp32/usb.c:96:9: error: implicit declaration of function 'tud_cdc_n_write_flush'; did you mean 'tinyusb_cdcacm_write_flush'? [-Werror=implicit-function-declaration]
>    96 |         tud_cdc_n_write_flush(CDC_ITF);
>       |         ^~~~~~~~~~~~~~~~~~~~~
>       |         tinyusb_cdcacm_write_flush
> /home/mickey-p/micropython/ports/esp32/usb.c: At top level:
> /home/mickey-p/micropython/ports/esp32/usb.c:41:16: warning: 'usb_rx_buf' defined but not used [-Wunused-variable]
>    41 | static uint8_t usb_rx_buf[CONFIG_TINYUSB_CDC_RX_BUFSIZE];
>       |                ^~~~~~~~~~
> cc1: some warnings being treated as errors
> [1285/1322] Building C object esp-idf/...-p/micropython/lib/littlefs/lfs2.c.objninja: build stopped: subcommand failed.
> ninja failed with exit code 1, output of the command is in the /home/mickey-p/micropython/ports/esp32/build/log/idf_py_stderr_output_12387 and /home/mickey-p/micropython/ports/esp32/build/log/idf_py_stdout_output_12387


Is it my fault with a building or anything else?

**Reply by robert-hh** [reply]
I do not have that here. Did you run `make submodules` after changing the branches?

**Reply by MikePolynin** [reply]
@robert-hh 
Sorry for waiting of the answer

If i try build for board=esp_32_generic_s3 it is ok
but i use devkit c 1 with n32r8v and try to build for board=esp_32_generic_s3 board_variant=spiram_oct

And this way i get error from my message above

**Reply by robert-hh** [reply]
No problem here with `make BOARD=ESP32_GENERIC_S3 BOARD_VARIANT=SPIRAM_OCT`. Maybe it's the esp-idf version. Micropython uses esp-idf v5.0.2. Are you sure that you updated the submodules?

**Reply by robert-hh** [reply]
The response here to `git submodule` is:
```
 84f56af13292d8f32c40acbd949bde698ddd4507 ../../lib/asf4 (remotes/origin/circuitpython)
 531cab9c278c947d268bd4c94ecab9153a961b43 ../../lib/axtls (axtls-2.1.5-44-g531cab9)
 35aaec4418ad78628a3b935885dd189d41ce779b ../../lib/berkeley-db-1.xx (heads/master-7-g35aaec4)
 77e752abd6a0992334047a48038a5a3960e5c6bc ../../lib/btstack (v1.4-2373-g77e752abd)
 2ab0db6e1fe9265fa9802a95f7f4d60b7f0d655a ../../lib/cyw43-driver (v1.0.1-9-g2ab0db6)
 e78939d32d1ccea9f0ba8bb42c51aceffd386b9b ../../lib/fsp (v4.4.0)
 e9de7e35f2339598b16cbb375f9992643ed81209 ../../lib/libffi (v3.2.1-146-ge9de7e3)
 5c5d513093075f7245ea522101b17c50aa579af2 ../../lib/libhydrogen (5c5d513)
 6ca936f6b588cee702c638eee75c2436e6cf75de ../../lib/lwip (STABLE-2_1_3_RELEASE)
 981743de6fcdbe672e482b6fd724d31d0a0d2476 ../../lib/mbedtls (v2.28.3)
 e025c843b60e93689f0f991d753010bb5bd6a722 ../../lib/micropython-lib (v0.1-1207-ge025c84)
 42849560ba7906f023f61e5f7ff3709ba2c1dfca ../../lib/mynewt-nimble (nimble_1_4_0_tag-2-g42849560)
 7a4c9d946cf1801771fc180acdbf7b878f270093 ../../lib/nrfx (v2.0.0)
 fa5a554c7944d2a196626f8d3631e44943f9abcc ../../lib/nxp_driver (list-13-gfa5a554)
 6a7db34ff63345a7badec79ebea3aaef1712f374 ../../lib/pico-sdk (1.5.1)
 abc67a11c6db271bedbb9f58be85d6f4e2ea8389 ../../lib/protobuf-c (v1.4.1)
 928df866e4d287ebc3c60726151513ebee609128 ../../lib/stm32lib (F0-1.9.0+F4-1.16.0+F7-1.7.0+G0-1.5.1+G4-1.3.0+H5-1.0.0+H7-1.11.0+L0-1.11.2+L1-1.10.3+L4-1.17.0+WB-1.10.0+WL-1.1.0-35-g928df86)
 1fdf29075d4e613eacfa881166015263797db0f6 ../../lib/tinyusb (0.13.0-1511-g1fdf29075)
 0803fc519ad7227e841287fb3638d6c8b2f111a1 ../../lib/wiznet5k (W5500_PHY_LINK-92-g0803fc5)
```

**Reply by MikePolynin** [reply]
In my case response to `git submodule` is almost the same:
>84f56af13292d8f32c40acbd949bde698ddd4507 lib/asf4 (remotes/origin/circuitpython)
 531cab9c278c947d268bd4c94ecab9153a961b43 lib/axtls (axtls-2.1.5-44-g531cab9)
 35aaec4418ad78628a3b935885dd189d41ce779b lib/berkeley-db-1.xx (heads/master-7-g35aaec4)
 77e752abd6a0992334047a48038a5a3960e5c6bc lib/btstack (v1.4-2373-g77e752abd)
 2ab0db6e1fe9265fa9802a95f7f4d60b7f0d655a lib/cyw43-driver (v1.0.2)
 e78939d32d1ccea9f0ba8bb42c51aceffd386b9b lib/fsp (v4.4.0)
 e9de7e35f2339598b16cbb375f9992643ed81209 lib/libffi (v3.2.1-146-ge9de7e3)
 5c5d513093075f7245ea522101b17c50aa579af2 lib/libhydrogen (5c5d513)
 6ca936f6b588cee702c638eee75c2436e6cf75de lib/lwip (STABLE-2_1_3_RELEASE)
 981743de6fcdbe672e482b6fd724d31d0a0d2476 lib/mbedtls (v2.28.3)
 e025c843b60e93689f0f991d753010bb5bd6a722 lib/micropython-lib (v0.1-1207-ge025c84)
 42849560ba7906f023f61e5f7ff3709ba2c1dfca lib/mynewt-nimble (nimble_1_4_0_tag-2-g42849560)
 7a4c9d946cf1801771fc180acdbf7b878f270093 lib/nrfx (v2.0.0)
 fa5a554c7944d2a196626f8d3631e44943f9abcc lib/nxp_driver (fa5a554)
 6a7db34ff63345a7badec79ebea3aaef1712f374 lib/pico-sdk (1.5.1)
 abc67a11c6db271bedbb9f58be85d6f4e2ea8389 lib/protobuf-c (v1.4.1)
 928df866e4d287ebc3c60726151513ebee609128 lib/stm32lib (F0-1.9.0+F4-1.16.0+F7-1.7.0+G0-1.5.1+G4-1.3.0+H5-1.0.0+H7-1.11.0+L0-1.11.2+L1-1.10.3+L4-1.17.0+WB-1.10.0+WL-1.1.0-35-g928df86)
 1fdf29075d4e613eacfa881166015263797db0f6 lib/tinyusb (0.15.0-765-g1fdf29075)
 0803fc519ad7227e841287fb3638d6c8b2f111a1 lib/wiznet5k (W5500_PHY_LINK-92-g0803fc5)

After resinstall esp-idf v5.1.1 i built with `make BOARD=ESP32_GENERIC_S3 BOARD_VARIANT=SPIRAM_OCT` without using menuconfig by manually changes in boards/esp32_generic_s3/sdconfig.board:
>CONFIG_ESPTOOLPY_FLASHMODE_OCT=y
CONFIG_ESPTOOLPY_FLASHFREQ_80M=y
CONFIG_ESPTOOLPY_AFTER_NORESET=y
CONFIG_ESPTOOLPY_FLASHSIZE_4MB=
CONFIG_ESPTOOLPY_FLASHSIZE_8MB=
CONFIG_ESPTOOLPY_FLASHSIZE_16MB=
CONFIG_ESPTOOLPY_FLASHSIZE_32MB=y
CONFIG_PARTITION_TABLE_CUSTOM=y
CONFIG_PARTITION_TABLE_CUSTOM_FILENAME="partitions-32MiB.csv"

and in boards/sdconfig.spiram_sx:
>MicroPython on ESP32-S2 and ESP32-PAD1_subscript_3, ESP IDF configuration with SPIRAM support
CONFIG_SPIRAM_MODE_OCT=y
CONFIG_SPIRAM_TYPE_AUTO=y
CONFIG_SPIRAM_CLK_IO=30
CONFIG_SPIRAM_CS_IO=26
CONFIG_SPIRAM_SPEED_80M=y
CONFIG_SPIRAM=y
CONFIG_SPIRAM_BOOT_INIT=y
CONFIG_SPIRAM_IGNORE_NOTFOUND=y
CONFIG_SPIRAM_USE_MALLOC=y
CONFIG_SPIRAM_MALLOC_ALWAYSINTERNAL=8192

After flasing i got error while running board:
>ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x7 (TG0WDT_SYS_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
Saved PC:0x40043ac8
SPIWP:0xee
Octal Flash Mode Enabled
For OPI Flash, Use Default Flash Boot Mode
mode:SLOW_RD, clock div:1
load:0x3c130020,len:0x412d0
load:0x3fc9ec00,len:0x4890
load:0x40374000,len:0xa488
ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x10 (RTCWDT_RTC_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
SPIWP:0xee
Octal Flash Mode Enabled
For OPI Flash, Use Default Flash Boot Mode
mode:SLOW_RD, clock div:1
load:0x3c130020,len:0x412d0
load:0x3fc9ec00,len:0x4890
load:0x40374000,len:0xa488
load:0x42000020,len:0x120878
Invalid image block, can't boot.
ets_main.c 329

**Reply by robert-hh** [reply]
Does the master branch build with these settings?
Did you try to build with esp-idf v5.0.2? When changing the idf version, please run `git submodule sync; git submodule update --init` in the esp-idf directory to update the submodules there.

**Reply by MikePolynin** [reply]
Hello

Trying to build with master branch

>git clone --recursive -b v5.1.1 https://github.com/espressif/esp-idf.git
cd ~/esp-idf
git submodule sync
git submodule update --init --recursive
cd ~/
git clone --recursive https://github.com/micropython/micropython.git
git submodule sync
git submodule update --init --recursive
cd ~/esp-idf
./install.sh esp32s3
. ./export.sh


manual changes:
in boards/esp32_generic_s3/sdconfig.board:
CONFIG_ESPTOOLPY_FLASHMODE_OCT=y
CONFIG_ESPTOOLPY_FLASHFREQ_80M=y
CONFIG_ESPTOOLPY_AFTER_NORESET=y
CONFIG_ESPTOOLPY_FLASHSIZE_4MB=
CONFIG_ESPTOOLPY_FLASHSIZE_8MB=
CONFIG_ESPTOOLPY_FLASHSIZE_16MB=
CONFIG_ESPTOOLPY_FLASHSIZE_32MB=y
CONFIG_PARTITION_TABLE_CUSTOM=y
CONFIG_PARTITION_TABLE_CUSTOM_FILENAME="partitions-32MiB.csv"

in boards/sdconfig.spiram_sx:
MicroPython on ESP32-S2 and ESP32-PAD1_subscript_3, ESP IDF configuration with SPIRAM support
CONFIG_SPIRAM_MODE_OCT=y
CONFIG_SPIRAM_TYPE_AUTO=y
CONFIG_SPIRAM_CLK_IO=30
CONFIG_SPIRAM_CS_IO=26
CONFIG_SPIRAM_SPEED_80M=y
CONFIG_SPIRAM=y
CONFIG_SPIRAM_BOOT_INIT=y
CONFIG_SPIRAM_IGNORE_NOTFOUND=y
CONFIG_SPIRAM_USE_MALLOC=y
CONFIG_SPIRAM_MALLOC_ALWAYSINTERNAL=8192


>make -C mpy-cross
cd ports/esp32
make BOARD=ESP32_GENERIC_S3 BOARD_VARIANT=SPIRAM_OCT submodules
make BOARD=ESP32_GENERIC_S3 BOARD_VARIANT=SPIRAM_OCT

Build is ok
After flashing the same result:
>ESP-ROM:esp32s3-20210327
Build:Mar 27 2021
rst:0x10 (RTCWDT_RTC_RST),boot:0x8 (SPI_FAST_FLASH_BOOT)
SPIWP:0xee
Octal Flash Mode Enabled
For OPI Flash, Use Default Flash Boot Mode
mode:SLOW_RD, clock div:1
load:0x3c120020,len:0x3c440
load:0x3fc9ec00,len:0x3bb0
load:0x42000020,len:0x117828
Invalid image block, can't boot.
ets_main.c 329

**Reply by MikePolynin** [reply]
If i try esp-idf 5.0.2 version - result is the same

**Reply by MikePolynin** [reply]
@robert-hh 
Hello

So i decided the problem with the building by using clean virtual machine.
As i find out problem was in flashing by Thonny. 

All submodules are fine.

I try build with a `make MICROPY_PY_NETWORK_WIZNET5K=5500 BOARD=ESP32_GENERIC_S3 BOARD_VARIANT=SPIRAM_OCT` and it is ok
but when i try run `nic = network.WIZNET5K(machine.SPI(1), pyb.Pin.board.X5, pyb.Pin.board.X4)`  after `import network`
i get an error `AttributeError: 'module' object has no attribute 'WIZNET5K'`

Can you give some advice please?

**Reply by robert-hh** [reply]
Thonny again!  It's not WIZNET5K, and your example is for the PybBoard V1.1. At an ESP32. A typical start sequence would be:
```
import network
import machine
import time
spi = machine.SPI(1, sck=machine.Pin(18), mosi=machine.Pin(21), miso=machine.Pin(19))
nic = network.LAN(phy_type=network.PHY_W5500, spi=spi, phy_addr=1,
                    cs=machine.Pin(5), int=machine.Pin(4))

nic.active(1)
while not nic.isconnected():
    print(".", end="")
    time.sleep(0.25)
```
You have to set the Pin numbers following the connections you have.

**Reply by robert-hh** [reply, marked as answer]
Meanwhile the W5500 branch has been merged into the master. So you do not have to use the PR, and from tomorrow on it should be available in the v1.22 preview build.

**Reply by MikePolynin** [reply]
Thank you very much for your answers and spending time!

My board now connect to the w55500

**Reply by B00mb0x** [reply]
I tried compiling the firmware using `make MICROPY_PY_NETWORK_WIZNET5K=5500 BOARD=ESP32_GENERIC_S3` and also tried adding `CONFIG_COMPILER_OPTIMIZATION_SIZE=y
CONFIG_ETH_SPI_ETHERNET_W5500=y` int sdkconfig.board as discussed here: https://github.com/orgs/micropython/discussions/11474
Both times the sample code from @robert-hh did run, but both times it got stuck printing dots. Looking with a Oscilloscope, I saw, that no SPI communication takes place. If I put in a `spi.write` statement, spi communication is taking place. Do you have any idea, what I could try next? I am using the micropython release 1.22.2 from February, so the PR should already be included.
